sphinx/api/built-in/rose_arch.rst

```diff
@@ -52,59 +81,130 @@ Example
+.. code-block:: rose
 
-   # Archive files matched by a glob to a directory
+   # A single glob
    [arch:worlds/]
    source=hello/worlds/*
 
-   # Archive multiple files matched by globs or names to a directory
+   # Three globs
    [arch:worlds/]
    source=hello/worlds/* greeting/worlds/* hi/worlds/*
 
-   # As above, but "greeting/worlds/*" may return an empty list
+Missing files and directories
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+It's also possibly to deal with a situation where one or more of the glob
+expressions might not return anything by putting brackets - ``()`` - around it:
+
+.. code-block:: rose
+
+   # If there isn't anything in greeting/worlds/ Rose Arch continues
    [arch:worlds/]
    source=hello/worlds/* (greeting/worlds/*) hi/worlds/*
 
-   # Target is optional, implied that sources may all be missing
+You can even tell Rose Arch that there may be nothing to archive, but to carry
+on:
+
+.. code-block:: rose
+
    [arch:(black-box/)]
    source=cats.txt dogs.txt
 
-   # Auto tar-gzip
+Zipping files
+^^^^^^^^^^^^^
+There are multiple ways of specifying that you want your achive to be
+compressed:
+
+You can infer compression from the target extension:
+
+.. code-block:: rose
+
+   [arch:planet.gz]
+   source=hello/planet.out
+
+...or manually specify a compression program. (In this case the ``out.gz`` is
+not regcognized by rose arch as an extension to be compressed.)
+
+.. code-block:: rose
+
+   [arch:planet.out.gz]
+   compress=gz
+   source=hello/planet.out
+
+For more details see :rose:conf:`rose_arch[arch]compress`
+
+Zipping directories
+^^^^^^^^^^^^^^^^^^^
+You can tar and zip entire directories - as with single files Rose Arch will
+attempt to infer archive and compression from ``[arch:TARGET.extension]`` if it
+can:
+
+.. code-block:: rose
+
    [arch:galaxies.tar.gz]
    source-prefix=hello/
    source=galaxies/*
    # File with multiple galaxies may be large, don't do its checksum
    update-check=mtime+size
 
+You might prefer to explicitly gzip each file in the source directory separately:
+
+.. code-block:: rose
+
    # Force gzip each source file
    [arch:stars/]
    source=stars/*
    compress=gzip
 
-   # Source name transformation
+Renaming files simply
+^^^^^^^^^^^^^^^^^^^^^
+You may wish to change the name of the archived files. By default the contents
+of your app'a ``[arch:Target]source=whatever-you-put-here`` and
+``$CYLC_TASK_CYCLE_TIME`` are available to you as python formatting strings
+``%(name)s`` and ``%(cycle)s``.
+
+.. code-block:: rose
+
    [arch:moons.tar.gz]
    source=moons/*
    rename-format=%(cycle)s-%(name)s
-   source-edit-format=sed 's/Hello/Greet/g' %(in)s >%(out)s
 
-   # Source name transformation with a rename-parser
-   [arch:unknown/stuff.pax]
-   rename-format=hello/%(cycle)s-%(name_head)s%(name_tail)s
-   rename-parser=^(?P<name_head>stuff)ing(?P<name_tail>-.*)$
-   source=stuffing-*.txt
+.. warning::
 
+   As ``%(cycle)s`` can be a path is may not always make sense to
```

Comment:
Eh, can `cycle` be a path, or is it that `name` is an absolute path?

Reply:
Yes. Good sanity there